### DISCUSSION TITLE
Getting typescript compile error on ring-client-api api.ts

### DISCUSSION THREAD

**Discussion by cardsfan7189** [dgreif/ring#1401] [Q&A] [answered]
I'm still a novice with Typescript.  I did a gitclone of this project, made no changes to the package files,  and encountering this compile error with ring-client-api/api.ts  Do I need to adjust my VS Code environment?

ring/packages/ring-client-api/api.ts:89:17 - error TS2347: Untyped function calls may not accept type arguments.

 89       } = await this.restClient.request<{
                    ~~~~~~~~~~~~~~~~~~~~~~~~~
 90         doorbots: CameraData[]
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
...
101         )[]
    ~~~~~~~~~~~
102       }>({ url: clientApi('ring_devices') }),
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

This is how I ran the compile:  tsc -t es5 tryit.ts

**Comment by tsightler** [comment, marked as answer]
What exactly is tryit.ts?  In general you do not need to compile this project to use it in another project since you should simply including the npm package as a dependency instead.

What happens if you attempt to compile only this project, i.e. switch to ring-client-api and just run tsc with no additional arguments?

**Reply by cardsfan7189** [reply]
tryit.ts may've been an unnecessary copy from one of the example scripts to something of my own.  I blew away the directory and did another git clone to start from scratch.  This time, went to the examples directory and did tsc api-example.ts without touching anything. Get the same error along with numerous others ../ring-client-api/api.ts:89:17 - error TS2347: Untyped function calls may not accept type arguments.

 89       } = await this.restClient.request<{
                    ~~~~~~~~~~~~~~~~~~~~~~~~~
 90         doorbots: CameraData[]
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
...
101         )[]
    ~~~~~~~~~~~
102       }>({ url: clientApi('ring_devices') }),
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

../ring-client-api/api.ts:330:52 - error TS2347: Untyped function calls may not accept type arguments.

330     const { user_locations: rawLocations } = await this.restClient.request<{
                                                       ~~~~~~~~~~~~~~~~~~~~~~~~~
331       user_locations: UserLocation[]
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
332     }>({ url: deviceApi('locations') })
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

../ring-client-api/api.ts:344:12 - error TS2347: Untyped function calls may not accept type arguments.

344     return this.restClient.request<any[]>({
               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
345       url: 'https://api.ring.com/integrations/amazonkey/v2/devices/lock_associations',
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
346     })
    ~~~~~~

../ring-client-api/api.ts:436:12 - error TS2347: Untyped function calls may not accept type arguments.

436     return this.restClient.request<ProfileResponse>({
               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
437       url: clientApi('profile'),
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
438     })
    ~~~~~~

**Reply by cardsfan7189** [reply]
sorry for the last reply. It was a waste of time and space.  I again blew away the directory created by the second git clone, but also saved the api-example.ts file.  Recreated a project directory, manually copied the tsconfig.json contents from github packages/tsconfig to the tsconfig.json in the project directory where the api-example.ts was also located. Ran tsc without any arguments like @tsightler said, and this time it worked.
